### ISSUE TITLE
Chemlight Management

### ISSUE THREAD

**Issue by joebayles** [Esri/squad-leader-android#98]
The only way to remove chemlights currently is to restart the application. There should be a way to remove/adjust chemlights from within the application.

Additionally, when Squad Leader is being used to demo (kiosk mode), it would be nice for the chemlites and other adjustments to wipe every time.

**Comment by jrweakland** [comment]
Backlog item: https://app.asana.com/0/23019535413944/23019535413977

**Comment by garys-esri** [comment]
I realize now we need to decide what happens on other users' displays when a user does any of the actions in the checklist above. In other words, if I modify or delete a chem light (or all chem lights) in Squad Leader, should messages go out to listening clients?

Here's what I think should happen.
- **Modify a chem light:** send an update message to listening clients. It would be silly to change a chem light just on the user's device.
- **Delete a chem light:** send a delete message to listening clients. The user intends to cancel that chem light for all users.
- **Delete all chem lights:** remove them just from the user's display but don't send out any messages. This capability is really for demo purposes. A Squad Leader user shouldn't be able to delete all chem lights for everyone using the platform. That would be done in a HQ or TOC if it ever needed to be done.
- **Process "remove" GeoMessage:** remove them just from the user's display but don't send out any messages. Our app just received the message and should not rebroadcast it.

If you disagree on any of this, please let me know soon.

**Comment by joebayles** [comment]
Modify - I agree.
Delete - I agree.
Delete all - Depends on the use case for chem-lites. If they are intended for "single-mission" use, I'd say it would be great to add a "decay" feature where they just go away after a specified amount of time. This is also a feature that is currently in place in FBCB2 for old SPOTREPs (Spot Reports). For instance, when I spot the enemy and submit that report, it will decay over time. It's represented graphically by transparency (it shouldn't gray out since color is an important part of the symbology representation). The other way to do it is to allow a delete all only for chemlites authored by that user.
Process "remove" - I agree.

Hope that makes sense.

**Comment by garys-esri** [comment]
The decay feature is a good idea. Let's create a separate issue for it, since it would be automated and the current issue is about the user managing chem lights, and also because we'll have to do the symbology some other way since ArcGIS Runtime's MessageProcessor won't do transparent symbols off the shelf.

I will implement the ability to delete chem lights created by the current user, i.e. during the current run of the application.

**Comment by joebayles** [comment]
:+1:

**Comment by garys-esri** [comment]
Update: I'm still working on this. Most of the code is done and I'm down to some minor bugs in my branch. I hope to finish this off today.

**Comment by garys-esri** [comment]
Fixed in #109. I asked @csmoore to review and merge if he is able.

**Comment by csmoore** [comment]
I guess this accidentally got auto-closed by one of the commit comments ("Fixes XXX" - since I didn't close this)...reopening so it can be verified

**Comment by garys-esri** [comment]
@csmoore, who would you like to close it then? It's assigned to you.

**Comment by csmoore** [comment]
It should be probably assigned to whoever is able to actually test/verify the issue (usually the originator, but not sure who is testing these issues on an android device).

**Comment by joebayles** [comment]
I cede my testing assignment to someone with more time this sprint... If there isn't anyone I'll figure it out.

**Comment by garys-esri** [comment]
@jrweakland is going to do it when the other issues for the upcoming release are addressed.

**Comment by jrweakland** [comment]
Reviewed at tested with Squad Leader 3.0.0 RC1 from https://esri.app.box.com/s/9hv7urwcvzisn67lyvfps855l8lediic

**Comment by garys-esri** [comment]
Thanks, @jrweakland. I removed the link to the file since it's internal and this repo is public. We will of course release the pre-built app on our downloads site once your testing is complete.
